### PR TITLE
User Guide

### PR DESCRIPTION
**Pull request by ywk253100**

### CODE REVIEW

File: docs/user_guide.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+#User Guide
+##Overview
+This guide takes you through the fundamentals of using Harbor. You'll learn how to use Harbor to:  
+
+* Manage your projects.
```
Comment (hainingzhang) [inline docs/user_guide.md]: remove all "." at the end

File: docs/user_guide.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+* Manage Harbor system if you are the system administrator.
+* Pull and push images using Docker client.
+
+
+##Role Based Access Control
+RBAC (Role Based Access Control) is provided in Harbor and there are four roles with different privileges:  
+
+* **Guest**: Guest has read-only privilege for a specified project.
+* **Developer**: Developer has read and write privileges for a project.
+* **ProjectAdmin**: When create a new project, you will be assigned the "ProjectAdmin" role to the project. Besides read-write privileges, the "ProjectAdmin" also has some management privileges, such as adding and removing members.
```
Comment (hainingzhang) [inline docs/user_guide.md]: When creating ...

File: docs/user_guide.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+* **SysAdmin**: "SysAdmin" has the most privileges. In addition to the privileges mentioned above, "SysAdmin" can also list all projects, set an ordinary user as administrator and delete users. The public project "library" is also owned by the administrator.  
+
+##Managing projects
+A project in Harbor is a namespace that role based access control is applied to it and the repositories in it. Two types of projects are provided:
```
Comment (hainingzhang) [inline docs/user_guide.md]: Replaced with below line:
A project in Harbor contains all repositories of an application. RBAC is applied to a project. There are two types of projects in Harbor:

File: docs/user_guide.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+* **Guest**: Guest has read-only privilege for a specified project.
+* **Developer**: Developer has read and write privileges for a project.
+* **ProjectAdmin**: When create a new project, you will be assigned the "ProjectAdmin" role to the project. Besides read-write privileges, the "ProjectAdmin" also has some management privileges, such as adding and removing members.
+* **SysAdmin**: "SysAdmin" has the most privileges. In addition to the privileges mentioned above, "SysAdmin" can also list all projects, set an ordinary user as administrator and delete users. The public project "library" is also owned by the administrator.  
+
+##Managing projects
+A project in Harbor is a namespace that role based access control is applied to it and the repositories in it. Two types of projects are provided: 
+
+* **Public**: All users have the read privilege to a public project, it's convenient for you to share some repositories with others in this way.
+* **Private**: A private project can only been accessed by its members.
```
Comment (hainingzhang) [inline docs/user_guide.md]: +* **Private**: A private project can only been accessed by users with proper privileges.

File: docs/user_guide.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+You can create a project after you signing in. Enable the "Public project" checkbox will make this project public.
```
Comment (hainingzhang) [inline docs/user_guide.md]: ....  after you signed in. .....

File: docs/user_guide.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+A project in Harbor is a namespace that role based access control is applied to it and the repositories in it. Two types of projects are provided: 
+
+* **Public**: All users have the read privilege to a public project, it's convenient for you to share some repositories with others in this way.
+* **Private**: A private project can only been accessed by its members.
+
+You can create a project after you signing in. Enable the "Public project" checkbox will make this project public.  
+
+![create project](img/create_project.png)  
+
+After the project being created, you can browse the repositories, members and access logs using the tabs in the left.
```
Comment (hainingzhang) [inline docs/user_guide.md]: +After the project is created, you can browser repositories, users and access logs using the navigation column on the left.

File: docs/user_guide.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+![create project](img/create_project.png)  
+
+After the project being created, you can browse the repositories, members and access logs using the tabs in the left.  
+
+![browse project](img/browse_project.png)  
+
+![browse project](img/project_log.png)  
+
+##Managing the members of a project
```
Comment (hainingzhang) [inline docs/user_guide.md]: remove "the"

File: docs/user_guide.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+![browse project](img/project_log.png)  
+
+##Managing the members of a project 
+###Adding members
+You can add members with different roles to an existing project.  
+
+![browse project](img/add_member.png)
+
+###Updating and removing members
+You can update or remove the members by click the icons.
```
Comment (hainingzhang) [inline docs/user_guide.md]: +You can update or remove a member by clicking the icon on the right.

File: docs/user_guide.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+![browse project](img/add_member.png)
+
+###Updating and removing members
+You can update or remove the members by click the icons.  
+
+![browse project](img/remove_update_member.png)
+
+##Searching projects and repositories
+If you have signed in, you will get a list of projects and repositories that you have read privilege to and contain the keywords. Otherwise, only the public projects will be listed.
```
Comment (hainingzhang) [inline docs/user_guide.md]: Entering a keyword in the search field at the top lists all matching projects and repos. The search result includes public repos and private repos you have access privilege to.

File: docs/user_guide.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+![browse project](img/search.png)
+
+##Administrator options
+###Setting administrator and deleting user
+Administrator can add "SysAdmin" role to an ordinary user and delete users by clicking the icons.  
+
+![browse project](img/set_admin_remove_user.png)
+
+##Pulling and pushing images using Docker client
+**NOTE: The Registry V2 is compatible only with Docker engine version 1.6.0 or higher, so the version of your Docker must meet the requirement.**
```
Comment (hainingzhang) [inline docs/user_guide.md]: +**NOTE: Harbor only supports Registry V2 API. You need to use Docker client 1.6.0 or higher.**

File: docs/user_guide.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+##Pulling and pushing images using Docker client
+**NOTE: The Registry V2 is compatible only with Docker engine version 1.6.0 or higher, so the version of your Docker must meet the requirement.**  
+###Pulling images
+If the project that the image belongs to is private, you should sign in first.  
+
+```sh
+$ docker login 10.117.169.182  
+```
+  
+And then you can pull the image.
```
Comment (hainingzhang) [inline docs/user_guide.md]: +You can now pull the image.

File: docs/user_guide.md [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+$ docker login 10.117.169.182  
+```
+  
+And then you can pull the image.  
+
+```sh
+$ docker pull 10.117.169.182/library/ubuntu:14.04  
+```
+
+**Note: Replace "10.117.169.182" with your Harbor IP or domain name.**
```
Comment (hainingzhang) [inline docs/user_guide.md]: +**Note: Replace "10.117.169.182" with the IP address or domain name of your Harbor node.**

File: docs/user_guide.md [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+
+###Pushing images
+Before pushing images, you must create corresponding projects using Harbor web UI.
```
Comment (hainingzhang) [inline docs/user_guide.md]: +Before pushing an image, you must create a corresponding project on Harbor web UI.

File: docs/user_guide.md [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+```sh
+$ docker pull 10.117.169.182/library/ubuntu:14.04  
+```
+
+**Note: Replace "10.117.169.182" with your Harbor IP or domain name.**
+
+###Pushing images
+Before pushing images, you must create corresponding projects using Harbor web UI. 
+
+Sign in first.
```
Comment (hainingzhang) [inline docs/user_guide.md]: +First, log in from Docker client:

File: docs/user_guide.md [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+###Pushing images
+Before pushing images, you must create corresponding projects using Harbor web UI. 
+
+Sign in first.  
+
+```sh
+$ docker login 10.117.169.182  
+```
+  
+Tag the image.
```
Comment (hainingzhang) [inline docs/user_guide.md]: +Tag the image:

File: docs/user_guide.md [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+$ docker login 10.117.169.182  
+```
+  
+Tag the image.  
+
+```sh
+$ docker tag ubuntu:14.04 10.117.169.182/demo/ubuntu:14.04  
+``` 
+
+Push the image.
```
Comment (hainingzhang) [inline docs/user_guide.md]: +Push the image:

File: docs/user_guide.md [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+
+```sh
+$ docker push 10.117.169.182/demo/ubuntu:14.04  
+```  
+
+**Note: Replace "10.117.169.182" with your Harbor IP or domain name.**
```
Comment (hainingzhang) [inline docs/user_guide.md]: +Note: Replace "10.117.169.182" with the IP address or domain name of your Harbor node.

File: docs/user_guide.md [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+
+
+##Role Based Access Control
+RBAC (Role Based Access Control) is provided in Harbor and there are four roles with different privileges:  
+
+* **Guest**: Guest has read-only privilege for a specified project.
+* **Developer**: Developer has read and write privileges for a project.
+* **ProjectAdmin**: When creating a new project, you will be assigned the "ProjectAdmin" role to the project. Besides read-write privileges, the "ProjectAdmin" also has some management privileges, such as adding and removing members.
+* **SysAdmin**: "SysAdmin" has the most privileges. In addition to the privileges mentioned above, "SysAdmin" can also list all projects, set an ordinary user as administrator and delete users. The public project "library" is also owned by the administrator.  
+
```
Comment (hainingzhang) [inline docs/user_guide.md]: * **Anonymous**: When a user is not logged in, the user is considered as an "anonymous" user. An anonymous user has no access to private projects and has read-only access to public projects.

File: docs/user_guide.md [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+##User account operations
+Username, Email, full name and password are required to sign up in Harbor. The password mast match the rule: at least 7 characters with 1 lowercase letter, 1 capital letter and 1 numeric character.
```
Comment (hainingzhang) [inline docs/user_guide.md]: A new user can sign up an account by going through the self-registration process. Username and email must be unique in the Harbor system. The password must contain at least 7 characters with 1 lowercase letter, 1 uppercase letter and 1 numeric character.

File: docs/user_guide.md [hunk 20]
```diff
@@ -0,0 +1,109 @@
...
+
+* **Guest**: Guest has read-only privilege for a specified project.
+* **Developer**: Developer has read and write privileges for a project.
+* **ProjectAdmin**: When creating a new project, you will be assigned the "ProjectAdmin" role to the project. Besides read-write privileges, the "ProjectAdmin" also has some management privileges, such as adding and removing members.
+* **SysAdmin**: "SysAdmin" has the most privileges. In addition to the privileges mentioned above, "SysAdmin" can also list all projects, set an ordinary user as administrator and delete users. The public project "library" is also owned by the administrator.  
+
+##User account operations
+Username, Email, full name and password are required to sign up in Harbor. The password mast match the rule: at least 7 characters with 1 lowercase letter, 1 capital letter and 1 numeric character.  
+  
+"Change Password" and "Forgot Password" are also supported. You can follow the steps to reset your password:
```
Comment (hainingzhang) [inline docs/user_guide.md]: When you forgot your password, you can follow the below steps to reset the password:

File: docs/user_guide.md [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+* **ProjectAdmin**: When creating a new project, you will be assigned the "ProjectAdmin" role to the project. Besides read-write privileges, the "ProjectAdmin" also has some management privileges, such as adding and removing members.
+* **SysAdmin**: "SysAdmin" has the most privileges. In addition to the privileges mentioned above, "SysAdmin" can also list all projects, set an ordinary user as administrator and delete users. The public project "library" is also owned by the administrator.  
+
+##User account operations
+Username, Email, full name and password are required to sign up in Harbor. The password mast match the rule: at least 7 characters with 1 lowercase letter, 1 capital letter and 1 numeric character.  
+  
+"Change Password" and "Forgot Password" are also supported. You can follow the steps to reset your password:  
+
+1. Click the link "forgot password" in the sign in page.
+2. Input the Email used when you signed up and a Email will be sent to it.
```
Comment (hainingzhang) [inline docs/user_guide.md]: 2. Input the email used when you signed up, an email will be sent out to you.

File: docs/user_guide.md [hunk 22]
```diff
@@ -0,0 +1,109 @@
...
+3. Click the link contained in the Email.
```
Comment (hainingzhang) [inline docs/user_guide.md]: 3. After receiving the email, click on the link in the email which directs you to a password reset web page.

File: docs/user_guide.md [hunk 23]
```diff
@@ -0,0 +1,109 @@
...
+4. Input your new password and click "Submit".
+
+
+##Managing projects
+A project in Harbor contains all repositories of an application. RBAC is applied to a project. There are two types of projects in Harbor:  
+
+* **Public**: All users have the read privilege to a public project, it's convenient for you to share some repositories with others in this way.
+* **Private**: A private project can only been accessed by users with proper privileges.
```
Comment (hainingzhang) [inline docs/user_guide.md]: ... can only be accessed ...

File: docs/user_guide.md [hunk 24]
```diff
@@ -0,0 +1,109 @@
...
+4. Input your new password and click "Submit".
+
+
+##Managing projects
+A project in Harbor contains all repositories of an application. RBAC is applied to a project. There are two types of projects in Harbor:  
+
+* **Public**: All users have the read privilege to a public project, it's convenient for you to share some repositories with others in this way.
+* **Private**: A private project can only been accessed by users with proper privileges.  
+
+You can create a project after you signed in. Enable the "Public project" checkbox will make this project public.
```
Comment (hainingzhang) [inline docs/user_guide.md]: ... Enabling the "Public project"...

File: docs/user_guide.md [hunk 25]
```diff
@@ -0,0 +1,109 @@
...
+
+You can create a project after you signed in. Enable the "Public project" checkbox will make this project public.  
+
+![create project](img/create_project.png)  
+
+After the project is created, you can browse repositories, users and access logs using the navigation column on the left.  
+
+![browse project](img/browse_project.png)  
+
+All access logs will be listed by clicking "Logs", you can filter them by username or operations and date which are supported in "Advanced Search".
```
Comment (hainingzhang) [inline docs/user_guide.md]: All access logs can be listed by clicking "Logs". You can apply a filter by username, or operations and dates under "Advanced Search".

File: docs/user_guide.md [hunk 26]
```diff
@@ -0,0 +1,109 @@
...
+![browse project](img/remove_update_member.png)
+
+##Searching projects and repositories
+Entering a keyword in the search field at the top lists all matching projects and repos. The search result includes public repos and private repos you have access privilege to.  
+
+![browse project](img/search.png)
+
+##Administrator options
+###Setting administrator and deleting user
+Administrator can add "SysAdmin" role to an ordinary user and delete users by clicking the icons.
```
Comment (hainingzhang) [inline docs/user_guide.md]: Administrator can add "SysAdmin" role to an ordinary user by toggling the switch under "System Admin". To delete a user, click on the recycle bin icon.

File: docs/user_guide.md [hunk 27]
```diff
@@ -0,0 +1,109 @@
...
+
+##Role Based Access Control
+RBAC (Role Based Access Control) is provided in Harbor and there are four roles with different privileges:  
+
+* **Guest**: Guest has read-only privilege for a specified project.
+* **Developer**: Developer has read and write privileges for a project.
+* **ProjectAdmin**: When creating a new project, you will be assigned the "ProjectAdmin" role to the project. Besides read-write privileges, the "ProjectAdmin" also has some management privileges, such as adding and removing members.
+* **SysAdmin**: "SysAdmin" has the most privileges. In addition to the privileges mentioned above, "SysAdmin" can also list all projects, set an ordinary user as administrator and delete users. The public project "library" is also owned by the administrator.  
+
+##User account operations
```
Comment (hainingzhang) [inline docs/user_guide.md]: ##User account

**Comment by reasonerjt** [comment]
LGTM, @ywk253100 could you please also update README.md in this PR to add reference to the user guide?

**Comment by ywk253100** [comment]
@reasonerjt  I have added it.

**Pull request merged**